python/nav/metrics/data.py

```diff
@@ -110,39 +110,55 @@ def get_metric_data(target, start="-5min", end="now"):
     if isinstance(end, datetime):
         end = end.strftime('%H:%M%Y%m%d')
 
-    query = {
-        'target': target,
-        'from': start,
-        'until': end,
-        'format': 'json',
-    }
-    query = urlencode(query, True)
-
-    _logger.debug("get_metric_data%r", (target, start, end))
-    req = Request(url, data=query.encode('utf-8'))
-    try:
-        response = urlopen(req)
-        json_data = json.load(codecs.getreader('utf-8')(response))
-        _logger.debug("get_metric_data: returning %d results", len(json_data))
-        return json_data
-    except HTTPError as err:
-        _logger.error(
-            "Got a 500 error from graphite-web when fetching %s" "with data %s",
-            err.url,
-            query,
-        )
-        _logger.error("Graphite output: %s", err.fp.read())
-        raise errors.GraphiteUnreachableError("{0} is unreachable".format(base), err)
-    except URLError as err:
-        raise errors.GraphiteUnreachableError("{0} is unreachable".format(base), err)
-    except ValueError:
-        # response could not be decoded
-        return []
-    finally:
+    def divide_chunks(list, n):
+        for i in range(0, len(list), n):
+            yield list[i : i + n]
```

Comment:
This implementation seems redundant. Check this out
https://github.com/Uninett/nav/blob/7fabaabb3c9f15d085ba55808f5ee0151c07b555/python/nav/util.py#L222-L230